routes/order_products.js

```diff
@@ -12,12 +19,49 @@ orderProductsRouter.post('/:orderId/products', requireUser, async (req, res, nex
         const addProduct = await addProductToOrder(orderId, productId, price, quantity)
         res.send({addProduct})
     } catch (error) {
-        
+        console.log(error)
     }
 })
 
 orderProductsRouter.patch('/order_products/:orderProductId', requireUser, async (req, res, next) => {
+    const { orderProductId } = req.params;
+    const { price, quantity } = req.body 
 
+    try {
+        const orderProduct = await getOrderProductById(id);
+        const order = await getOrderById(orderProduct.orderId)
+        const user = await getUserById(order.userId)
+    if (req.user.id === order.userId) {
+        const updatedOrderProduct = await updateOrderProduct({id: Number(orderProductId), price, quantity})
+        if(updatedOrderProduct) {
+        res.send(updatedOrderProduct)
+        }
+    } else {
+        throw new Error("Only the user can update their order")
+        }
+    } catch (error) {
+        console.log(error)
```

Comment:
Anywhere inside a ROUTE (not other stuff), inside the `catch` block, call `next(error)` instead of just console.log.

Reply:
Doneskies